### PR TITLE
Use load tests protocol to run the test suite

### PR DESCRIPTION
**Pull request by s-t-e-v-e-n-k**
Since we no longer require Python 2, we can make use of the load tests protocol and automatic discovery to find all tests we want to run, rather than constructing a list of TestSuites by hand. As a consequence, I've had to fiddle with two test classes so they are not discovered and run. Drive-by the tox environment list as well.

### CODE REVIEW

**Comment by s-t-e-v-e-n-k** [comment]
Fixes #31 which I forgot to mention in the commit message.

File: tests/test_compat.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def itervalues(self):
         return iter(self.values())
 
 
-class TestCaseBase(unittest.TestCase):
+class BaseTestCase:
```
Comment (jirihnidek) [inline tests/test_compat.py:39]: I would not rename the name of the class. There is no reason for that.

This class still should be child of `unittest.TestCase`. It is strange that this class is currently not child of `unittest.TestCase`, but it uses many methods from this class. e.g. `self.assertTrue()`

Maybe it will not be possible to use automatic discovery of two test cases (child classes of `TestCaseBase`)

File: runtests.py
Comment (jirihnidek) [inline runtests.py:1]: Is this file actually necessary? When I use py.test, then it is possible to drop this file completely.

**Review by jirihnidek — CHANGES_REQUESTED** [review]
Hi and thanks for the PR. I have some requests and suggestions. It also seems that repository should contain `cfgparser.1` with following content:

```ini
[Foo Bar]
foo = newbar
```

File: runtests.py
Comment (s-t-e-v-e-n-k) [reply runtests.py:1]: If you run `pytest` directly, I'm not certain if the doctests are run.

File: tests/test_compat.py (hunk 1 above)
Comment (s-t-e-v-e-n-k) [reply tests/test_compat.py:39]: If it starts with Test* it will be collected, and if this runs, it causes failures because `config_class` is not set.

Comment (jirihnidek) [reply tests/test_compat.py:39]: The class should be subclass of `unittest.TestCase`, then it would be collected anyway. Maybe, if `config_class` is set to `None`, then it could be used for condition for skipping tests. It could be possible to use `unittest.skipIf`.

File: runtests.py
Comment (jirihnidek) [reply runtests.py:1]: Hmm, `tox` still requires this file :-/ Keep it.

**Review by jirihnidek — APPROVED** [review]
Thanks for the update :+1: . LGTM now
